### PR TITLE
Reminder email

### PR DESCRIPTION
**Pull request by peymanmortazavi**

### CODE REVIEW

File: app/user_auth.py [hunk 1]
```diff
@@ -14,10 +14,37 @@
 from lib.database.reservations import get_reservation_for_user
 from lib.database.promotion_keys import pop_promotion_key
 from lib.database.showtimes import get_showtime
+from .lib.email_sender import send_reminder
```
Comment (mynameisfiber) [inline app/user_auth.py]: this won't import.... look 4 lines below for an example on how to import from `email_sender`

File: app/user_auth.py [hunk 2]
```diff
@@ -14,10 +14,37 @@
...
 
+@secured
+class UserReminder(BaseHandler):
+    _ioloop = ioloop.IOLoop().instance()
+
+    @web.asynchronous
+    @gen.coroutine
+    def get(self):
+        all_reservations = yield get_reservations()
+        all_showtimes = yield get_showtimes()
```
Comment (mynameisfiber) [inline app/user_auth.py]: remove this

File: app/user_auth.py [hunk 3]
```diff
@@ -14,10 +14,37 @@
...
+        for reservation in all_reservations:
+            show_id = reservation['showtime_id']
+            date_str = filter(
```
Comment (mynameisfiber) [inline app/user_auth.py]: remove this and have `show_meta = yield get_showtime(show_id)` and the string date is in there. No need to resolve problems we've already solved.

File: app/user_auth.py [hunk 4]
```diff
@@ -14,10 +14,37 @@
...
+                (lambda x: x['id'] == show_id),
+                all_showtimes)[0]
+            user_id = reservation['user_id']
+            user = get_user(user_id)
```
Comment (mynameisfiber) [inline app/user_auth.py]: this won't work since get_user is a coroutine

**Comment by wannabeCitizen** [comment]
tested and ready for review

**Comment by mynameisfiber** [comment]
I just pushed up changed to how the code got the show date.... we have a function `get_showtime` that's less flimsy than that filter code.

My one worry is that this one function sends out a fuckton of e-mails.... chances are going to get blocked or rate limited.  That being, the send_mail takes like 4 seconds to run for me on just 2 or 3 e-mail addresses.  Maybe we should have better logging so that if it does fail, we know who we've sent the e-mail to already and we don't spam them.

**Comment by wannabeCitizen** [comment]
I'm not sure it'll fail on our end? I'm pretty sure that if an email fails it'll show up in the gmail client, but I'm not sure if we get any error from the `send_email()` function. I know when we send to a bad email address that error doesn't come through on the server, just shows up in the actual gmail client.

**Comment by mynameisfiber** [comment]
so you think that gmail won't rate limit us for sending ~200 emails in quick succession? it definitely does fail on our end if gmail no longer authorizes us.

**Comment by wannabeCitizen** [comment]
I'm honestly not sure what they'll do. We could put a delay in the script if we want? I've never seen the error before but if we want to pickle email addresses before a failure, that'd be fine with me.

**Comment by mynameisfiber** [comment]
I would wager that `smtplib` throws one of these exceptions https://docs.python.org/3/library/smtplib.html#smtplib.SMTPException

I say do either (although preferably both):
- [ ] add a `yield gen.sleep(1)` between each send
- [ ] add a `logger.info("Sending e-mail to: ".....)` between each send

**Comment by wannabeCitizen** [comment]
Alright, made it so we log errors and keep track of which emails actually happened. Want to review quickly?

**Pull request merged**